test/python/transpiler/test_barrier_before_final_measurements.py

```diff
@@ -392,6 +393,29 @@ def test_conditioned_on_single_bit(self):
         pass_ = BarrierBeforeFinalMeasurements()
         self.assertEqual(expected, pass_(circuit))
 
+    def test_output_deterministic(self):
+        """Test that the output barriers have a deterministic ordering (independent of
+        PYTHONHASHSEED).  This is important to guarantee that any subsequent topological iterations
+        through the circuit are also deterministic; it's in general not possible for all transpiler
+        passes to produce identical outputs across all valid topological orderings, especially if
+        those passes have some stochastic element."""
+        measure_order = list(range(20))
+        random.Random(2023_02_10).shuffle(measure_order)
+        circuit = QuantumCircuit(20, 20)
+        circuit.barrier([5, 2, 3])
+        circuit.barrier([7, 11, 14, 2, 4])
+        circuit.measure(measure_order, measure_order)
+
+        # All the barriers should get merged together.
+        expected = QuantumCircuit(20, 20)
+        expected.barrier(range(20))
+        expected.measure(measure_order, measure_order)
+
+        output = BarrierBeforeFinalMeasurements()(circuit)
```

Comment:
Did you want to add a direct test of `MergeAdjacentBarriers` or do you think the high level via `BarrierBeforeFinalMeasurements` is sufficient coverage.

Reply:
Oh duh, I should have done this. I'd originally thought it was just an internal implementation detail because I didn't spot its tests, but I can see them now.  Added in fbfada661aa3c273f2f80566a48e301d2767a9b2.